### DISCUSSION TITLE
Webinterface doesn't start automatically

### DISCUSSION THREAD

**Discussion by FlixXxer1984** [JaapvanEkris/openrowingmonitor#37] [Q&A] [answered]
Hey,
that's a great project. At the beginning of the year I bought a "WaterRowing-Machine" manufactured by Crivit and sold via Lidl in Germany. It was relatively easy to connect the sensor to my Raspberry Pi 3b.
Installing the software was hardly more difficult. Following the instructions in the other Thread ("Update or Installation v1beta_updates ") works very fine. The Webinterface was available from my Notebook and also from different Smartphones. The Displays shows , wenn I pull the "WaterRower".
So far so good...today I want to adjust the settings to optimize the displayed or calculated values. So I boot the Raspberry and want to access the Website. But my browser Firefox show an error that the IP-Adress is not accessible.
Similtanous a ping is possible, also a ssh-connection via putty an acces via SMB via Windows Explorer.
First I started the updateopenrowingmonitor.sh (after I called
`sudo git config --global --add safe.directory /opt/openrowingmonitor`
This was followed by
`sudo apt-get update`
and
`sudo apt-get upgrade`
and
`sudo npm install npm -g`
Now npm is installed in v10.4.0
Connecting via ssh reports that rowingmonitor 6.1.21-v8+ #1642 is running.

But the Webinterface is still not available via Webbrowser. Is it possible to start this service manually?
Or can somebody do some troubleshooting? I'm not really a Linux-Nerd nor a Pro in using GitHub. So please be patient :-)

Thanks a lot for your support!

**Comment by Gordon-Shumway2** [comment]
Please see 

    https://github.com/JaapvanEkris/openrowingmonitor/blob/v1beta_updates/docs/installation.md

especially the commands 

    sudo systemctl status openrowingmonitor

and 

    sudo journalctl -u openrowingmonitor

Their output should tell you about any errors during startup. Fell free to upload the output. 

You mentioned "Connecting via ssh reports that rowingmonitor 6.1.21-v8+ #1642 is running."  ssh usually will not tell you such a thing, so I wonder how you got that information. Can you elaborate? 
Update: my bad, I thought you were referring to an ORM version of "6.1.21". You were rather referring to the OS version running on the Pi. 

best regards, 
Michael aka Gordon

P.S.: did you learn about ORM on mydealz.de?

**Reply by FlixXxer1984** [reply]
Thank you very much. First, yes I do. I've read your posting in this thread on mydealz :-) And what should I say? You got me and I wanted to try this individual way.

But back to topic:
`sudo systemctl status openrowingmonitor`
tells
`openrowingmonitor.service - Open Rowing Monitor
     Loaded: loaded (/lib/systemd/system/openrowingmonitor.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Thu 2024-02-08 19:21:10 CET; 18min ago
   Main PID: 830 (code=exited, status=1/FAILURE)
        CPU: 3.231s

Feb 08 19:21:09 rowingmonitor systemd[1]: openrowingmonitor.service: Consumed 3.231s CPU time.
Feb 08 19:21:10 rowingmonitor systemd[1]: openrowingmonitor.service: Scheduled restart job, restart counter is at 5.
Feb 08 19:21:10 rowingmonitor systemd[1]: Stopped Open Rowing Monitor.
Feb 08 19:21:10 rowingmonitor systemd[1]: openrowingmonitor.service: Consumed 3.231s CPU time.
Feb 08 19:21:10 rowingmonitor systemd[1]: openrowingmonitor.service: Start request repeated too quickly.
Feb 08 19:21:10 rowingmonitor systemd[1]: openrowingmonitor.service: Failed with result 'exit-code'.
Feb 08 19:21:10 rowingmonitor systemd[1]: Failed to start Open Rowing Monitor.`

Can you explain this failure or how to fix it?

My first try with ORM has ended similar before. As result I reinstalled the OS and ORM via the command in the thread referred above. I'm sure that sounds funny and nearly every user says something like that, but I'm sure that I have not changed anything in the configuration ;-)

As an addition a part of the result of "sudo journalctl -u openrowingmonitor":

Feb 03 21:04:01 rowingmonitor systemd[1]: Started Open Rowing Monitor.
Feb 03 21:04:04 rowingmonitor npm[5657]: > openrowingmonitor@0.9.0 start
Feb 03 21:04:04 rowingmonitor npm[5657]: > node app/server.js
Feb 03 21:04:09 rowingmonitor npm[5677]: ==== Open Rowing Monitor 0.9.0 ====
Feb 03 21:04:09 rowingmonitor npm[5677]: bluetooth profile: FTMS Rower
Feb 03 21:04:09 rowingmonitor npm[5677]: Starting a just row session, no time or distance target set
Feb 03 21:04:09 rowingmonitor npm[5677]: heart rate profile: Off
Feb 03 21:04:09 rowingmonitor npm[5677]: ant plus profile: Off
Feb 03 21:04:09 rowingmonitor npm[5677]: webserver running on port 80
Feb 03 21:04:10 rowingmonitor npm[5700]: Gpio-service: pin number 17, polling interval 5 us, triggered on Up flank, minimal pulse time 50 us
Feb 03 21:32:36 rowingmonitor npm[5677]: websocket client connected
Feb 03 21:32:45 rowingmonitor npm[5677]: websocket client disconnected
Feb 03 23:43:27 rowingmonitor npm[5677]: websocket client connected
Feb 04 00:18:28 rowingmonitor npm[5677]: SIGTERM signal was received, shutting down gracefully
Feb 04 00:18:28 rowingmonitor npm[5677]: shutting down all peripherals
Feb 04 00:18:28 rowingmonitor npm[5700]: 2024-02-04 00:18:28 sigHandler: Unhandled signal 18, terminating
Feb 04 00:18:28 rowingmonitor systemd[1]: Stopping Open Rowing Monitor...
Feb 04 00:18:28 rowingmonitor npm[5700]: 2024-02-04 00:18:28 sigHandler: Unhandled signal 15, terminating
Feb 04 00:18:28 rowingmonitor npm[5700]: 2024-02-04 00:18:28 sigHandler: Unhandled signal 18, terminating
Feb 04 00:18:28 rowingmonitor npm[5700]: 2024-02-04 00:18:28 sigHandler: Unhandled signal 11, terminating
Feb 04 00:18:28 rowingmonitor npm[5700]: 2024-02-04 00:18:28 sigHandler: Unhandled signal 11, terminating
Feb 04 00:18:28 rowingmonitor systemd[1]: openrowingmonitor.service: Succeeded.
Feb 04 00:18:28 rowingmonitor systemd[1]: Stopped Open Rowing Monitor.
Feb 04 00:18:28 rowingmonitor systemd[1]: openrowingmonitor.service: Consumed 15min 53.735s CPU time.
-- Boot 66e8eef1533342e8964f1446957b07a6 --
Feb 04 00:18:45 rowingmonitor systemd[1]: Started Open Rowing Monitor.
Feb 04 00:18:50 rowingmonitor npm[760]: > openrowingmonitor@0.9.0 start
Feb 04 00:18:50 rowingmonitor npm[760]: > node app/server.js
Feb 04 00:18:51 rowingmonitor npm[785]: file:///opt/openrowingmonitor/config/rowerProfiles.js:210
Feb 04 00:18:51 rowingmonitor npm[785]:         numOfImpulsesPerRevolution: 2,
Feb 04 00:18:51 rowingmonitor npm[785]:                                   ^
Feb 04 00:18:51 rowingmonitor npm[785]: SyntaxError: Unexpected token ':'
Feb 04 00:18:51 rowingmonitor npm[785]:     at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:152:18)
Feb 04 00:18:51 rowingmonitor npm[785]:     at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:298:14)
Feb 04 00:18:51 rowingmonitor npm[785]: Node.js v18.19.0
Feb 04 00:18:51 rowingmonitor systemd[1]: openrowingmonitor.service: Main process exited, code=exited, status=1/FAILURE
Feb 04 00:18:51 rowingmonitor systemd[1]: openrowingmonitor.service: Failed with result 'exit-code'.
Feb 04 00:18:51 rowingmonitor systemd[1]: openrowingmonitor.service: Consumed 3.868s CPU time.
Feb 04 00:18:51 rowingmonitor systemd[1]: openrowingmonitor.service: Scheduled restart job, restart counter is at 1.
Feb 04 00:18:51 rowingmonitor systemd[1]: Stopped Open Rowing Monitor.
Feb 04 00:18:51 rowingmonitor systemd[1]: openrowingmonitor.service: Consumed 3.868s CPU time.
Feb 04 00:18:52 rowingmonitor systemd[1]: Started Open Rowing Monitor.
Feb 04 00:18:54 rowingmonitor npm[796]: > openrowingmonitor@0.9.0 start
Feb 04 00:18:54 rowingmonitor npm[796]: > node app/server.js
Feb 04 00:18:54 rowingmonitor npm[808]: file:///opt/openrowingmonitor/config/rowerProfiles.js:210
Feb 04 00:18:54 rowingmonitor npm[808]:         numOfImpulsesPerRevolution: 2,
Feb 04 00:18:54 rowingmonitor npm[808]:                                   ^
Feb 04 00:18:54 rowingmonitor npm[808]: SyntaxError: Unexpected token ':'
Feb 04 00:18:54 rowingmonitor npm[808]:     at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:152:18)
Feb 04 00:18:54 rowingmonitor npm[808]:     at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:298:14)
Feb 04 00:18:54 rowingmonitor npm[808]: Node.js v18.19.0
Feb 04 00:18:54 rowingmonitor systemd[1]: openrowingmonitor.service: Main process exited, code=exited, status=1/FAILURE
Feb 04 00:18:54 rowingmonitor systemd[1]: openrowingmonitor.service: Failed with result 'exit-code'.
Feb 04 00:18:54 rowingmonitor systemd[1]: openrowingmonitor.service: Consumed 3.487s CPU time.
Feb 04 00:18:55 rowingmonitor systemd[1]: openrowingmonitor.service: Scheduled restart job, restart counter is at 2.
Feb 04 00:18:55 rowingmonitor systemd[1]: Stopped Open Rowing Monitor.

**Reply by Abasz** [reply]
Malformed config . Please see:

npm[808]: file:///opt/openrowingmonitor/config/rowerProfiles.js:210
Feb 04 00:18:54 rowingmonitor npm[808]: numOfImpulsesPerRevolution: 2,
Feb 04 00:18:54 rowingmonitor npm[808]: ^
Feb 04 00:18:54 rowingmonitor npm[808]: SyntaxError: Unexpected token ':'
Feb 04 00:18:54 rowingmonitor npm[808]: at ModuleLoader.moduleStrategy (node:internal/modules/esm/transl

**Comment by Gordon-Shumway2** [comment]
From "sudo journalctl -u openrowingmonitor":

> Feb 04 00:18:51 rowingmonitor npm[785]: file:///opt/openrowingmonitor/config/rowerProfiles.js:210
Feb 04 00:18:51 rowingmonitor npm[785]: numOfImpulsesPerRevolution: 2,
Feb 04 00:18:51 rowingmonitor npm[785]: ^
Feb 04 00:18:51 rowingmonitor npm[785]: SyntaxError: Unexpected token ':'
Feb 04 00:18:51 rowingmonitor npm[785]: at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:152:18)
Feb 04 00:18:51 rowingmonitor npm[785]: at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:298:14)
Feb 04 00:18:51 rowingmonitor npm[785]: Node.js v18.19.0
Feb 04 00:18:51 rowingmonitor systemd[1]: openrowingmonitor.service: Main process exited, code=exited, status=1/FAILURE

There seems to be a syntax errror on line 210 within file `/opt/openrowingmonitor/config/rowerProfiles.js`. 
Can you upload that file so I can have a look at it? 

Regards, 
Michael

Edit: Darn... Abasz beat me by a minute. Need to type faster :)

**Comment by FlixXxer1984** [comment]
Arg...I think it would be much more helpful if I show you the newest log since the last reboot...
The following is just the last block of the Log.

-- Boot a38e9ff6b8e84dbd85a551decee40800 --
Feb 08 22:08:25 rowingmonitor systemd[1]: Started Open Rowing Monitor.
Feb 08 22:08:40 rowingmonitor npm[864]: > openrowingmonitor@0.9.0 start
Feb 08 22:08:40 rowingmonitor npm[864]: > node app/server.js
Feb 08 22:08:41 rowingmonitor npm[880]: node:internal/modules/cjs/loader:1473
Feb 08 22:08:41 rowingmonitor npm[880]:   return process.dlopen(module, path.toNamespacedPath(filename));
Feb 08 22:08:41 rowingmonitor npm[880]:                  ^
Feb 08 22:08:41 rowingmonitor npm[880]: Error: The module '/opt/openrowingmonitor/node_modules/@abandonware/bluetooth-hci-socket/lib/binding/bluetooth_hci_socket.node'
Feb 08 22:08:41 rowingmonitor npm[880]: was compiled against a different Node.js version using
Feb 08 22:08:41 rowingmonitor npm[880]: NODE_MODULE_VERSION 108. This version of Node.js requires
Feb 08 22:08:41 rowingmonitor npm[880]: NODE_MODULE_VERSION 115. Please try re-compiling or re-installing
Feb 08 22:08:41 rowingmonitor npm[880]: the module (for instance, using `npm rebuild` or `npm install`).
Feb 08 22:08:41 rowingmonitor npm[880]:     at Module._extensions..node (node:internal/modules/cjs/loader:1473:18)
Feb 08 22:08:41 rowingmonitor npm[880]:     at Module.load (node:internal/modules/cjs/loader:1207:32)
Feb 08 22:08:41 rowingmonitor npm[880]:     at Module._load (node:internal/modules/cjs/loader:1023:12)
Feb 08 22:08:41 rowingmonitor npm[880]:     at Module.require (node:internal/modules/cjs/loader:1235:19)
Feb 08 22:08:41 rowingmonitor npm[880]:     at require (node:internal/modules/helpers:176:18)
Feb 08 22:08:41 rowingmonitor npm[880]:     at Object.<anonymous> (/opt/openrowingmonitor/node_modules/@abandonware/bluetooth-hci-socket/lib/native.js:6:15)
Feb 08 22:08:41 rowingmonitor npm[880]:     at Module._compile (node:internal/modules/cjs/loader:1376:14)
Feb 08 22:08:41 rowingmonitor npm[880]:     at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
Feb 08 22:08:41 rowingmonitor npm[880]:     at Module.load (node:internal/modules/cjs/loader:1207:32)
Feb 08 22:08:41 rowingmonitor npm[880]:     at Module._load (node:internal/modules/cjs/loader:1023:12) {
Feb 08 22:08:41 rowingmonitor npm[880]:   code: 'ERR_DLOPEN_FAILED'
Feb 08 22:08:41 rowingmonitor npm[880]: }
Feb 08 22:08:41 rowingmonitor npm[880]: Node.js v20.11.0
Feb 08 22:08:41 rowingmonitor systemd[1]: openrowingmonitor.service: Main process exited, code=exited, status=1/FAILURE
Feb 08 22:08:41 rowingmonitor systemd[1]: openrowingmonitor.service: Failed with result 'exit-code'.
Feb 08 22:08:41 rowingmonitor systemd[1]: openrowingmonitor.service: Consumed 4.949s CPU time.
Feb 08 22:08:41 rowingmonitor systemd[1]: openrowingmonitor.service: Scheduled restart job, restart counter is at 5.
Feb 08 22:08:41 rowingmonitor systemd[1]: Stopped Open Rowing Monitor.
Feb 08 22:08:41 rowingmonitor systemd[1]: openrowingmonitor.service: Consumed 4.949s CPU time.
Feb 08 22:08:41 rowingmonitor systemd[1]: openrowingmonitor.service: Start request repeated too quickly.
Feb 08 22:08:41 rowingmonitor systemd[1]: openrowingmonitor.service: Failed with result 'exit-code'.
Feb 08 22:08:41 rowingmonitor systemd[1]: Failed to start Open Rowing Monitor.

[ORM Config Files.zip](https://github.com/JaapvanEkris/openrowingmonitor/files/14214726/ORM.Config.Files.zip)


I just added the 3 config files, so you can look at them. The rowerprofiles.js doesn't have a line 210 anymore. I replaced the content with the rowerprofiles.js from the GitHub-Ressource.

**Reply by Abasz** [reply]
I think you need to run `npm ci` in the orm directory. 

The issue is that npm install was issed while an older version of node was installed.

**Comment by Gordon-Shumway2** [comment, marked as answer]
Your latest issues were most likely caused by installing npm which broke library dependencies:

> Feb 08 22:08:41 rowingmonitor npm[880]: Error: The module '/opt/openrowingmonitor/node_modules/@abandonware/bluetooth-hci-socket/lib/binding/bluetooth_hci_socket.node'
> Feb 08 22:08:41 rowingmonitor npm[880]: was compiled against a different Node.js version using
> Feb 08 22:08:41 rowingmonitor npm[880]: NODE_MODULE_VERSION 108. This version of Node.js requires
> Feb 08 22:08:41 rowingmonitor npm[880]: NODE_MODULE_VERSION 115. Please try re-compiling or re-installing 

My recommendation would be to invest 30 minutes of your and your computer's life time and start from scratch, following the documented installation instructions. IE write a new OS onto the SD card and run 


        /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/JaapvanEkris/openrowingmonitor/v1beta_updates/install/install.sh)"

as outlined in 

       https://github.com/JaapvanEkris/openrowingmonitor/blob/v1beta_updates/docs/installation.md

That will take about 30 minutes, throw some warnings one should ignore and put your ORM installation on the same level as ours regarding libraries and npm versions. That in turn makes it a lot easier to troubleshoot any issues.

If you already made some configuration changes to those three files you might want to back them up first and use them within the new installation.  

Kind regards, 
Michael

**Comment by FlixXxer1984** [comment]
Hey Abasz, hey Gordon aka Alf aka Michael,
thanks a lot for your bugfixing-attemps. Finally I reinstalled OS and reinstall ORM with the command Michael posted above.
Now everything is running fine again as before.

Just two more questions:
1. Is it neccessary to reboot the Pi everytime when I change something in the config.js or in the rowerprofiles.js?
    Update: As I read on
[https://github.com/JaapvanEkris/openrowingmonitor/blob/v1beta_updates/docs/rower_settings.md]
that should not be neccessary.
3. Assumed that I found some kind of working configuration respectively working parameters, what should I do with that findings?

Best greetings and have a good weekend!

**Reply by JaapvanEkris** [reply]
when you change settings, you can do "sudo systemctl restart openrowingmonitor"

You can post the settings here, along with a raw recording of at least 20 strokes, and the anticipated numbers (distance, time). If we have that, we can maintain them as a supported device (and update the settings if we change the engine).
